### ISSUE TITLE
Download of models is broken - can't get logging to provide more details due to /dev/stderr logging

### ISSUE THREAD

**Issue by GiantCrocodile** [nextcloud/recognize#956]
### Which version of recognize are you using?

4.3.2

### Enabled Modes

Object recognition, Face recognition, Video recognition, Music recognition

### TensorFlow mode

Normal mode

### Downstream App

other, please specify below

### Which Nextcloud version do you have installed?

27.0.2

### Which Operating system do you have installed?

Ubuntu 22 LTS

### Which database are you running Nextcloud on?

postgresql

### Which Docker container are you using to run Nextcloud? (if applicable)

23.0.6

### How much RAM does your server have?

not relevant

### What processor Architecture does your CPU have?

x86_64

### Describe the Bug

Hello,

I try to get the Recognize app to work on my Cloudron.io setup. Currently I'm running NextCloud 27.0.2 with latest Recognize app release. Installation was fine but my setup is stuck at downloading the models. I don't see any indication that models are being downloaded and it's broken since days. I tried to get more information from the logging but Cloudron.io does log to `/dev/stderr` (can't be changed) and even with debug level zero I don't get any more information except this:

`{"reqId":"<removed>","level":3,"time":"2023-08-20T09:55:38+00:00","remoteAddr":"<removed>","user":"--","app":"PHP","method":"GET","url":"/settings/admin/recognize","message":"fopen(/dev/stderr): Failed to open stream: No such file or directory at /app/code/lib/private/Log/File.php#84","userAgent":"<removed>","version":"27.0.2.1","data":{"app":"PHP"}}`

So I think logging is either broken or not providing more details and I don't know how to retrigger the download to get more details. I suspect the installation fails because some path might be read-only in my setup.

Can you provide more insight on how to debug or fix this? Thank you.

Best Regards

### Expected Behavior

Download of models work

### To Reproduce

just install the app and wait hours or days. You will still get an warning that the download of the models is not done and thus they are missing.

### Debug log

no further details beside of what I wrote above

**Comment by marcelklehr** [comment]
Yes, seems like nextcloud logging is broken, I can't help with that. As for the model download, have you tried the download-models command as documented in the recognize admin panel?

**Comment by GiantCrocodile** [comment]
I didn't see the download-models command there. This was what I was hoping/looking for! I get these errors:

```
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:50+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chown(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2199","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:50+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chgrp(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2200","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:50+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chown(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2199","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:50+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chgrp(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2200","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:50+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chown(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2199","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:50+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chgrp(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2200","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:50+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chown(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2199","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:50+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chgrp(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2200","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:50+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chown(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2199","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:50+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chgrp(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2200","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chown(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2199","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chgrp(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2200","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chown(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2199","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chgrp(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2200","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chown(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2199","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chgrp(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2200","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chown(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2199","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chgrp(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2200","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chown(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2199","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chgrp(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2200","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chown(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2199","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chgrp(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2200","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chown(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2199","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chgrp(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2200","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chown(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2199","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chgrp(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2200","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chown(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2199","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chgrp(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2200","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chown(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2199","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chgrp(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2200","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chown(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2199","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
{"reqId":"Z0GFigprSVnyNfrdxbLu","level":3,"time":"2023-08-20T19:28:51+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"chgrp(): Operation not permitted at /app/code/3rdparty/pear/archive_tar/Archive/Tar.php#2200","userAgent":"--","version":"27.0.2.1","data":{"app":"PHP"}}
```

**Comment by marcelklehr** [comment]
Well, your the unix user that runs php needs to be able to write files in the nextcloud directory

**Comment by GiantCrocodile** [comment]
I'm not sure this is a general write permission issue. I could imagine it's just a very specific path that is read-only. Where are the models stored? In general I'm able to upload documents and edit them in NextCloud. Furthermore I'm able to install the Recognize app itself without write permission issues.

**Comment by marcelklehr** [comment]
Recognize wants to download the models into /nextcloud/apps/recognize/models

**Comment by GiantCrocodile** [comment]
I just checked the path you have mentioned and I see that the models are stored there for some reason now. I don't know how this comes. NextCloud didn't update, just Cloudron and Recognize app. I don't see how a Cloudron update should have solved this so I think some change in the Recognize app fixed this issue for me. I verified this in the NextCloud settings as well: It says that the models were downloaded successfully.
